conformance/tests/httproute-missing-reference-policy.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteMissingReferencePolicy)
```

Comment:
That definitely makes sense for most cases and would be a reason to keep this as a separate test from the successful cross-namespace ref using `ReferencePolicy`, I just wasn't sure about the [exception](https://gateway-api.sigs.k8s.io/v1alpha2/api-types/referencepolicy/#exceptions) case - if an implementation falling in that category should show this test as failing with an annotation/explanation or be able to skip it?

Reply:
Good call on that exception. Maybe we should be introducing two separate but related concepts here. One would be `SupportedFeatures`, the other would be `Exemptions`, both would behave similarly. Each test could define a list of Features and/or Exemptions. If the test runner did not have a Feature from the `supportedFeatures` list specified or had an Exemption from the `exemptions` list specified, the test would be skipped.

Related comment: We should probably rename that test now that we'll have 2 tests related to HTTPRoute references crossing namespace boundaries in invalid ways. Maybe `httproute-invaliid-cross-namespace-backend-ref` and `httproute-invaliid-cross-namespace-parent-ref`?

Reply:
Though I'm not sure how frequently we expect this to be used, [ReferencePolicy exceptions](https://gateway-api.sigs.k8s.io/v1alpha2/api-types/referencepolicy/#exceptions) includes this note which adds to the need for exceptions here:
> There are some situations where it MAY be acceptable to ignore ReferencePolicy in favor of some other security mechanism. This MAY only be done if other mechanisms like NetworkPolicy can effectively limit cross-namespace references by the implementation.